### PR TITLE
fix(FX-3314): Don't send an empty request until `index_name` is selected

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Fix**

This PR resolves [FX-3314]

### Description
PR includes a check that doesn't send a request to Algolia until the `indexName` is selected for which the search results should be displayed

#### Before
https://user-images.githubusercontent.com/3513494/133645831-d3a6b4ec-5cd9-447d-9651-380fbfd5367c.mp4

#### After
https://user-images.githubusercontent.com/3513494/133645906-3229c503-8998-4c3c-8b85-7a30097d688b.mp4

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests for my changes, or my changes don't require testing, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/master/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/master/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Don't send an empty request until `index_name` is selected - dimatretyak

</details>

[FX-3314]: https://artsyproduct.atlassian.net/browse/FX-3314

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Don't send an empty request until `index_name` is selected - dimatretyak)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 698c8080c214e69daf985c426694c4ebc61d5c0a
</p>

File: src/lib/utils/useAlgoliaClient.ts [hunk 1]
```diff
@@ -1,11 +1,33 @@
-import algoliasearch, { AlgoliaSearchOptions, SearchClient } from "algoliasearch"
+import { AlgoliaSearchOptions, SearchClient } from "algoliasearch"
+import algoliasearch from "algoliasearch/lite"
```
Comment (gkartalis) [inline src/lib/utils/useAlgoliaClient.ts:2]: Was this intentional?
Are there any differences between algoliasearch imported from `"algoliasearch/lite"` vs `"algoliasearch"`?

File: src/lib/utils/useAlgoliaClient.ts [hunk 2]
```diff
@@ -1,11 +1,33 @@
...
 
 export const useAlgoliaClient = (appID: string, apiKey: string, options?: AlgoliaSearchOptions) => {
   const [searchClient, setSearchClient] = useState<SearchClient | null>(null)
 
   useEffect(() => {
-    const client = algoliasearch(appID, apiKey, options)
+    const algoliaClient = algoliasearch(appID, apiKey, options)
+    const client: SearchClient = {
+      ...algoliaClient,
+      // @ts-ignore
```
Comment (gkartalis) [inline src/lib/utils/useAlgoliaClient.ts:12]: I looked into how we can avoid the ts-ignore here but didn't find any straightforward solution.

**Review by gkartalis — APPROVED** [review]

File: src/lib/utils/useAlgoliaClient.ts (hunk 1 above)
Comment (dimatretyak) [reply src/lib/utils/useAlgoliaClient.ts:2]: Nope

Comment (dimatretyak) [reply src/lib/utils/useAlgoliaClient.ts:2]: Everything was done according to [official instructions](https://www.algolia.com/doc/guides/building-search-ui/going-further/conditional-requests/react/)

**Pull request merged**